h/models/annotation.py

```diff
@@ -1,20 +1,32 @@
 import datetime
+from enum import Enum
 from uuid import UUID
 
 import sqlalchemy as sa
 from sqlalchemy.dialects import postgresql as pg
 from sqlalchemy.ext.hybrid import hybrid_property
 from sqlalchemy.ext.mutable import MutableDict, MutableList
+from sqlalchemy.orm import Mapped, relationship
 
 from h.db import Base, types
 from h.models.group import Group
 from h.util import markdown_render, uri
 from h.util.user import split_user
 
 
+class ModerationStatus(Enum):
+    APPROVED = "APPROVED"
+    PENDING = "PENDING"
+    DENIED = "DENIED"
+    SPAM = "SPAM"
```

Comment:
I posted [a comment on the POC](https://github.com/hypothesis/h/pull/9461#discussion_r2037301818) about whether we want to use strings in the DB here or just ints?

Reply:
This type of enum is stored in the DB as an integer. I reckon this is the right way way to do "translate strings to integers" when you do not care about the actual integer value.

If we wanted to just rename a value we might not bother on the DB or we could alter the type:

`ALTER TYPE name RENAME value `denied` to `rejected`;` 

that would just change the type and no rows would need to be updated.